### ISSUE TITLE
redirect old language URL to language index page

### ISSUE THREAD

**Issue by xrotwang** [clld/phoible#11]
If we switch to relating inventories to Glottolog languoids (dialects and languages), we must switch the vafriety identifiers to Glottocodes, thus breaking the old variety URLs. Since varieties have not been considered core resources of PHOIBLE Online, it seems best to redirect the old URLs to the languages index page, rather than returning HTTP 410 (which would be more semantically correct).

**Comment by drammock** [comment]
So the idea is that the content for Aja (Benin) at http://phoible.org/languages/ajg would move to http://phoible.org/languages/ajab1235, and the old URL would redirect to the languages index page?

I realize that in some cases one ISO 639-3 based URL may correspond to multiple new Glottocode-based URLs... so in that case is it possible for old ISO-based URLs to automatically return search results for that code?  So if ISO code `xxx` corresponds to glottocodes `yyyy1234` and `zzzz5678`, then the URL `phoible.org/languages/xxx` would return a page with links to the `yyyy1234` and `zzzz5678` pages.  Is that easy? hard? undesirable for some reason I don't recognize?

**Comment by xrotwang** [comment]
Redirecting to a page listing all languages mapped to a given ISO code should be possible.

**Comment by drammock** [comment]
Great, thanks @xrotwang. @bambooforest any opinion here?

**Comment by bambooforest** [comment]
I think this solution sounds good.

**Comment by xrotwang** [comment]
Just tried to implement this redirection, but would like to avoid the intermediate page listing possible new matches for simplicity. This would mean the following ISO codes don't have a matching new language (the ones prefixed with `---` have no match, the other ones have multiple matches; so 22 out of 1,672 would end up with HTTP 410):
```
+++ izi {'izii1239', 'ikwo1238'}
--- ori
--- jar
--- adc
+++ kjn {'oyka1239', 'kunj1245'}
--- noo
--- nbf
--- dap
--- gio
--- gny
--- baz
--- qpt
--- psx
+++ sgw {'chah1248', 'seba1251', 'ezha1238'}
--- xuu
--- bsc
+++ gup {'guma1252', 'naia1238', 'gunw1252', 'mura1269', 'gune1238'}
--- stc
--- yrm
--- gmo
+++ din {'dink1262', 'sout2832', 'nort2815'}
+++ nyf {'kamb1298', 'kaum1238'}
```

acceptable?

**Comment by xrotwang** [comment]
ping @bambooforest

**Comment by bambooforest** [comment]
@xrotwang we definitely updated some of the ISO codes given their changes

**Comment by xrotwang** [comment]
So given that there are also changes, I'd say we can live with breaking 22 URLs - of pages which are not representing core data of PHOIBLE, but rather serve as navigation.

**Comment by drammock** [comment]
breaking 22 URLs is OK with me, but note that some of these could be redirected (the ones where there is only one glottocode in PHOIBLE that matches the current ISO codes that replaced the missing one). Examples (not a complete list):

- `ori` [changed from language to macrolanguage](https://iso639-3.sil.org/request/2011-159), with constituent members Oriya `ory` and Sambalpuri `spv`. Only `ory` is in PHOIBLE (`oriy1255`)
- `noo` [split](https://iso639-3.sil.org/request/2010-033) into `dtd` and `nuk`. Only `nuk` is in PHOIBLE (`nuuc1236`)
- `dap` [split](https://iso639-3.sil.org/request/2011-132) into `tgj` and `njz`. Only `njz` is in PHOIBLE (`nyis1236`)
- `gio` [split](https://iso639-3.sil.org/request/2011-054) into several new ISO codes, but of them only `gqu` is in PHOIBLE (`qaua1234`)

Others cannot be redirected easily, e.g.:

- `nbf` [split](https://iso639-3.sil.org/request/2010-023) into `nxq` and `nru`. PHOIBLE has one of each (`naxi1245`=`nxq`, `yong1270`=`nru`) so no obvious redirect choice here.

Is it worth it to go through the remaining ones one by one to determine if a single redirect target makes sense?  (note: even if we do this, I'm viewing it as a deprecation cycle; those URLs should go dead after the next release)

**Comment by xrotwang** [comment]
Yes, we can go through these one by hand. I'll leave the issue open. Once we've gone through the list, the glottocode we'd like to redirect to can go in [this JSON](https://github.com/clld/phoible/blob/master/replacements.json) - from where I can pick them up when running a DB update.
